les/api.go

```diff
@@ -24,121 +24,121 @@ import (
 func NewPrivateLesServerAPI(server *LesServer) *PrivateLesServerAPI {
-	vip := &vipClientPool{
-		clients: make(map[enode.ID]vipClientInfo),
-		totalBw: server.totalBandwidth,
-		pm:      server.protocolManager,
+	priority := &priorityClientPool{
+		clients:  make(map[enode.ID]priorityClientInfo),
+		totalCap: server.totalCapacity,
+		pm:       server.protocolManager,
 	}
-	server.protocolManager.vipClientPool = vip
+	server.protocolManager.priorityClientPool = priority
 	return &PrivateLesServerAPI{
-		server: server,
-		pm:     server.protocolManager,
-		vip:    vip,
+		server:   server,
+		pm:       server.protocolManager,
+		priority: priority,
 	}
 }
 
-// TotalBandwidth queries total available bandwidth for all clients
-func (api *PrivateLesServerAPI) TotalBandwidth() hexutil.Uint64 {
-	return hexutil.Uint64(api.server.totalBandwidth)
+// TotalCapacity queries total available capacity for all clients
+func (api *PrivateLesServerAPI) TotalCapacity() hexutil.Uint64 {
+	return hexutil.Uint64(api.server.totalCapacity)
 }
 
-// MinimumBandwidth queries minimum assignable bandwidth for a single client
-func (api *PrivateLesServerAPI) MinimumBandwidth() hexutil.Uint64 {
-	return hexutil.Uint64(api.server.minBandwidth)
+// MinimumCapacity queries minimum assignable capacity for a single client
+func (api *PrivateLesServerAPI) MinimumCapacity() hexutil.Uint64 {
+	return hexutil.Uint64(api.server.minCapacity)
 }
 
-// vipClientPool stores information about prioritized clients
-type vipClientPool struct {
-	lock                                  sync.Mutex
-	pm                                    *ProtocolManager
-	clients                               map[enode.ID]vipClientInfo
-	totalBw, totalVipBw, totalConnectedBw uint64
-	vipCount                              int
+// priorityClientPool stores information about prioritized clients
+type priorityClientPool struct {
+	lock                                          sync.Mutex
+	pm                                            *ProtocolManager
+	clients                                       map[enode.ID]priorityClientInfo
+	totalCap, totalpriorityCap, totalConnectedCap uint64
+	priorityCount                                 int
 }
 
-// vipClientInfo entries exist for all prioritized clients and currently connected free clients
-type vipClientInfo struct {
-	bw        uint64 // zero for non-vip clients
+// priorityClientInfo entries exist for all prioritized clients and currently connected free clients
+type priorityClientInfo struct {
+	cap       uint64 // zero for non-priority clients
 	connected bool
-	updateBw  func(uint64)
+	updateCap func(uint64)
 }
 
-// SetClientBandwidth sets the priority bandwidth assigned to a given client.
-// If the assigned bandwidth is bigger than zero then connection is always
-// guaranteed. The sum of bandwidth assigned to priority clients can not exceed
-// the total available bandwidth.
+// SetClientCapacity sets the priority capacity assigned to a given client.
```

Comment:
This explanation is a bit confusing for me. What does it mean to assign the capacity of a client? Sounds almost like we're setting the _priority_ of the client?

Reply:
Priority has a slightly different meaning in my terminology. Being a priority client means you will never be disconnected and you can even connect if a non-priority client has to be kicked out to make room for you. Priorities can be compared but they are not scalar values. Capacity, on the other hand, is a scalar value that is proportional to the amount of requests served per time unit. It makes sense to buy priority even if you buy the same capacity as a free client gets (but you get it guaranteed for a long time).
Btw flow and capacity are established terms in graph theory and I think they can be applied here.
https://en.wikipedia.org/wiki/Flow_network